types/atom/index.d.ts

```diff
@@ -1,17 +1,23 @@
-// Type definitions for Atom 1.20
+// Type definitions for Atom 1.21
 // Project: https://github.com/atom/atom
 // Definitions by: GlenCFL <https://github.com/GlenCFL>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
 /// <reference types="node" />
-/// <reference types="electron" />
 /// <reference types="jquery" />
 /// <reference types="atom-keymap" />
 /// <reference types="event-kit" />
 /// <reference types="first-mate" />
 /// <reference types="text-buffer" />
 
+// Atom is built on (and thus bundles) Electron. We replicate that here.
+// This file can be obtained from the Electron releases page:
+//   https://github.com/electron/electron/releases/
+// See Atom's package.json file in order to determine the correct version to
+// bundle.
+/// <reference path="./electron.d.ts" />
```

Comment:
Electron provides its own `.d.ts`.
I can't agree this method. I think it is  better to have less maintenance effort.

Reply:
This isn't your typical usage case for Electron. Those definitions are gated behind the dependencies, all of which are completely unnecessary with Atom. Bundling Electron definitions in this manner requires <5 minutes of maintenance each time Atom increases the Electron version, which only happens every 4 months or so. I don't really agree that this is too much maintenance effort in order to get rid of those dependencies.

The concern about the dependencies of ```@types/atom``` is due to [atom-ts-transpiler](https://github.com/smhxx/atom-ts-transpiler). We're going to be increasing installation time for everyone using a package developed utilizing both this transpiler and ```@types/atom```, with the two going hand-in-hand really.

I'm going to strip the ElectronWindow reference for either an ```object``` reference or a neutered version of ElectronWindow. The latter would ironically require much more maintenance.